### ISSUE TITLE
frcli nodereport - payment htlc has a route with zero hops

### ISSUE THREAD

**Issue by githorray** [lightninglabs/faraday#49]
```
$ frcli nodereport
[frcli] rpc error: code = Unknown desc = payment htlc has a route with zero hops
```
My node is quite old and has some early payments with no hops stored.

I had a similar problem with joule that was fixed.

https://github.com/joule-labs/joule-extension/pull/207

Versions of faraday and lnd:

frcli version 0.2.0-alpha commit=v0.2.0-alpha-9-gdf39c5a5c9e27d6935d14db1f1c0e7ad2197acc4
lnd version 0.10.99-beta commit=clock/v1.0.0-66-ged93c16e51abfd3512cb9a2800811b598f6768cd

**Comment by carlaKC** [comment]
Thanks for reporting! 

We're currently looking at payment hops to detect whether we have made a payment or ourselves (in the hope of identifying circular rebalances). We needed to look at hops rather than payment request because we don't always have payment request present, but I think it would make sense to fall back to checking the payment request for a destination in this case. If we have neither, I think logging a warning and just accepting that some old circular payments may not be detected is ok. 

In the meantime, you should be able to get around the issue by running `nodereport` with a specific start date that is after this payment.

**Comment by carlaKC** [comment]
> period too long for granularity level

Ah, I know what's going on here. Will get a fix up asap. Basically the api we use to get fiat estimates from has a tradeoff between accuracy of data and the length of time you query for. It's currently hardcoded which isn't good, I'll update it to shift this "granularity" with the length of query. 

>  Unknown desc = payment htlc has a route with zero hops

I am intrigued that you have a payment with no htlcs which has zero hops from the last month. Have you recently upgraded? Also going to put the fix in for this, but I am curious.

**Comment by githorray** [comment]
I have been recklessly running master since launching the node. :)

**Comment by carlaKC** [comment]
If you know which payment is tripping you up, could you find it in `listpayments` for me and post the output? I just want to make sure that the fix I put up covers this case. 

I don't need payment hash/amount or any identifiable fields, so feel free to remove that info. Specifically interested in the payment status, and the list of htlcs (if it's there). 
```
"status": "?",
"htlcs":   [list of htlcs, if present],
```

**Comment by githorray** [comment]
The last payment with an empty set of hops is

```
"SUCCEEDED"
"1531397351000000000"
{
  "total_time_lock": 0,
  "total_fees": "0",
  "total_amt": "X000",
  "hops": [],
  "total_fees_msat": "0",
  "total_amt_msat": "X000000"
}

```

**Comment by carlaKC** [comment]
Awesome, thanks for the output! Fix is in #53, I'll update the issue once it's merged. 

You'll see a warning `payment: {hash} destination unknown" in your logs once the fix is merged. For these payments where we don't know the destination (because they have no hops), we might not pick up circular payments properly (they'll just look like regular payments). Other than that, the report will be unaffected.

**Comment by githorray** [comment]
I tested https://github.com/lightninglabs/faraday/pull/53. I can pull a default node report now.
